### ISSUE TITLE
FAIL: Expected minimum docker-compose version to be 1.24.1 but found Docker Compose version 2.5.0

### ISSUE THREAD

**Issue by jamincollins** [getsentry/self-hosted#1550]
### Version

any version prior to 21.5.0

### Steps to Reproduce

Any `docker-compose` 2.x release installed (essentially any current `docker-compose`).

```bash
cd ${self-host-clone}
docker-compose down
git fetch --all
git checkout 21.5.0
./install.sh
```

### Expected Result

Normal upgrade would happen.

### Actual Result

upgrade fails with:

```
FAIL: Expected minimum docker-compose version to be 1.24.1 but found Docker Compose version 2.5.0
```

In order to proceed, the check must be corrected or commented out:
```bash
$ git diff
diff --git a/install/check-minimum-requirements.sh b/install/check-minimum-requirements.sh
index 4527a22..ba764c9 100644
--- a/install/check-minimum-requirements.sh
+++ b/install/check-minimum-requirements.sh
@@ -20,10 +20,10 @@ if [[ "$(ver $DOCKER_VERSION)" -lt "$(ver $MIN_DOCKER_VERSION)" ]]; then
   exit 1
 fi
 
-if [[ "$(ver $COMPOSE_VERSION)" -lt "$(ver $MIN_COMPOSE_VERSION)" ]]; then
-  echo "FAIL: Expected minimum docker-compose version to be $MIN_COMPOSE_VERSION but found $COMPOSE_VERSION"
-  exit 1
-fi
+#if [[ "$(ver $COMPOSE_VERSION)" -lt "$(ver $MIN_COMPOSE_VERSION)" ]]; then
+#  echo "FAIL: Expected minimum docker-compose version to be $MIN_COMPOSE_VERSION but found $COMPOSE_VERSION"
+#  exit 1
+#fi
 
 if [[ "$CPU_AVAILABLE_IN_DOCKER" -lt "$MIN_CPU_HARD" ]]; then
   echo "FAIL: Required minimum CPU cores available to Docker is $MIN_CPU_HARD, found $CPU_AVAILABLE_IN_DOCKER"
```

**Comment by ethanhs** [comment]
Hi!

Could you please provide what the output of `$(ver $COMPOSE_VERSION)` and `$(ver MIN_COMPOSE_VERSION)` (you can just add e.g. `echo $(ver $COMPOSE_VERSION)` above/in place of the if check).

I tested the versions given in the error and on my MacOS laptop 2.5.0 did not compare less than 1.24.1. I will have to test on a Linux machine in a bit to double check this isn't some bash difference or something.

**Comment by jamincollins** [comment]
```
$ git diff
diff --git a/install/check-minimum-requirements.sh b/install/check-minimum-requirements.sh
index 4527a22..59847dc 100644
--- a/install/check-minimum-requirements.sh
+++ b/install/check-minimum-requirements.sh
@@ -20,6 +20,10 @@ if [[ "$(ver $DOCKER_VERSION)" -lt "$(ver $MIN_DOCKER_VERSION)" ]]; then
   exit 1
 fi
 
+echo "$(ver $COMPOSE_VERSION)"
+echo "$(ver $MIN_COMPOSE_VERSION)"
+echo "$(ver $COMPOSE_VERSION)" -lt "$(ver $MIN_COMPOSE_VERSION)"
+
 if [[ "$(ver $COMPOSE_VERSION)" -lt "$(ver $MIN_COMPOSE_VERSION)" ]]; then
   echo "FAIL: Expected minimum docker-compose version to be $MIN_COMPOSE_VERSION but found $COMPOSE_VERSION"
   exit 1
```

```
$ ./install.sh 
▶ Parsing command line ...

▶ Setting up error handling ...

▶ Checking minimum requirements ...
option '--no-ansi' is DEPRECATED ! Please use '--ansi' instead.
0006000
1024001
0006000 -lt 1024001
FAIL: Expected minimum docker-compose version to be 1.24.1 but found Docker Compose version 2.6.0
```

**Comment by ethanhs** [comment]
Could you tell me what OS/version you are on? It seems that awk is behaving differently than expected. Perhaps we should consider using a version check based on `sort -V`.

**Comment by jamincollins** [comment]
Arch Linux, fully up to date.

```
$ docker-compose version
Docker Compose version 2.6.0
```

```
$ awk --version
GNU Awk 5.1.1, API: 3.1 (GNU MPFR 4.1.0-p13, GNU MP 6.2.1)
Copyright (C) 1989, 1991-2021 Free Software Foundation.

This program is free software; you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation; either version 3 of the License, or
(at your option) any later version.

This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.

You should have received a copy of the GNU General Public License
along with this program. If not, see http://www.gnu.org/licenses/.
```

**Comment by ethanhs** [comment]
Also is there a reason you are tied to versions before 21.5.0? Could you try with the latest?

**Comment by jamincollins** [comment]
I've successfully managed to go through the upgrade and I'm now running the current version. Just filing this bug report for completeness and to document the issue.

**Comment by jamincollins** [comment]
I should be able to test in a bit and get you the output of the above commands in my environment.

**Comment by aminvakil** [comment]
We cannot fix what has been released in versions prior to 21.5.0 right now, and I think we can suggest (not to support :smiley:) any **new user** to install versions newer than 1 year ago.

"Also is there a reason you are tied to versions before 21.5.0? Could you try with the latest?"

**Comment by ethanhs** [comment]
I think I am going to close this. I haven't been able to reproduce the issue and we are changing the code for dealing with version checks anyway: https://github.com/getsentry/self-hosted/pull/1553

I appreciate the bug report all the same @jamincollins :)
